react/AddToCartUrl.tsx

```diff
@@ -13,38 +21,45 @@ const enforceArray = (x: any) => {
 }
 
 const AddToCartUrl: FunctionComponent = () => {
-  const [addToCart] = useMutation(AddToCart)
+  const { addItem } = useOrderItems()
 
   // This ensures the checkout cookie exists
-  const { loading } = useQuery(OrderForm, { ssr: false })
+  const { loading } = useQuery(OrderFormQuery, { ssr: false })
 
   useEffect(() => {
-    if (!loading) {
-      const { sku, seller, qty } = map(enforceArray, {
-        sku: undefined,
-        seller: undefined,
-        qty: undefined,
-        ...parse(window.location && window.location.search),
-      })
+    if (loading) {
+      return
+    }
 
-      const newItems = []
-      for (let i = 0; i < sku.length; i++) {
-        newItems.push({
-          id: sku[i],
-          seller: seller[i] || 1,
-          quantity: qty[i] || 1,
-        })
-      }
-
-      addToCart({
-        variables: {
-          items: newItems,
-        },
-      }).then(() => window.location.replace('/cart'))
+    // parses all the query string parameters (in form of "sku=123,seller=1,qty=2")
+    // and group them by SKU, to be added to the cart later.
+    const { sku, seller, qty } = Object.entries({
+      sku: undefined,
+      seller: undefined,
+      qty: undefined,
+      ...parse(window.location?.search),
+    })
+      .map(([key, value]) => [key, enforceArray(value)] as const)
+      .reduce<{ sku: string[]; seller: number[]; qty: number[] }>(
+        (obj, [key, value]) => ({ ...obj, [key]: value }),
+        { sku: [], seller: [], qty: [] }
+      )
```

Comment:
Does this work? Feels a bit easier to understand to me.
```suggestion
      .reduce<{ sku: string[]; seller: number[]; qty: number[] }>(
        (obj, [key, value]) => ({ ...obj, [key]: enforceArray(value) }),
        { sku: [], seller: [], qty: [] }
      )
```

Reply:
hmmm, I think it works too